addons/web_editor/static/src/scss/web_editor.frontend.scss

```diff
@@ -35,6 +35,35 @@
     @extend %o-we-background-layer;
 
     @each $module, $shapes in $o-bg-shapes {
+        @each $shape, $style in $shapes {
+            $url-params: '';
+            $colors: map-get($style, 'colors');
+            @each $i in $colors {
+                $cc: 5 - $i;
+                @if ($cc == 0) {
+                    $cc: 5;
+                }
```

Comment:
General comment about the PR: I think you don't have to duplicate any code. Keep the same map of shapes as before (no `$o-bg-shapes-compat`), just add a new attribute alongside `colors` in each shape to mention the new cc colors. And then in the loop here you generate the old class for the "colors" attribute (if any, not required for new shapes in the future) and the new class for the new attribute (required).

Reply:
@sdegueldre What I meant was for an entry of the shapes map to be like this:
```scss
'Airy/01': ('position': bottom, 'size': 100% auto, 'cc-colors': (4), 'colors': (1), 'repeat-y': false)
```
No mapping between old colors and cc colors: you "add a new attribute alongside colors in each shape to mention the new cc colors". If that attribute is there you use the mentioned cc bg color to generate the _cc classes and/or if the old  `colors` attribute is there you use it to generate the old classes. For old shapes, we use both attributes. For new shapes we'll only use the new one.

And maybe one day we'll have information about user colors during migration and we could migrate old classes to new classes :shrug: 

Unrelated question: will this fix be ok in master for the system which guess to color to use on a shape being connected to a previous shape?

Reply:
Note: I chose cc-colors quickly. Maybe just cc-bgs or color-combination-bgs makes more sense?

Reply:
I understood what you meant, the problem is: I still need to map from a color number (c1) to a different cc (c1 => cc4), if I only have the cc color I'd need to map that to the color number so that I can use it the the URL params.

So if I need to convert it anyway, I'm not sure there's any point in duplicating this info and creating a dichotomy of colors vs cc-colors in the shape declaration. I'm not sure what the benefit is, it also makes creating new shapes less natural: at the moment, the colors field in this dict is simply "which `o-color-x`s of the default palette does this shape use"

This does mean that in some palettes (currently I think there are only 3) one of the colors that you specify is not actually used directly, but instead the bg-color of the cc that usually has the equivalent palette color as a background is used. This could be confusing in rare cases, but in that case maybe it makes more sense to still only have one color field and deduce the old color number from the cc number.

In any case, as pointed out I will still need to have both cc and color number since I need the color number in the url unless we want to change the shape system to be based on `DEFAULT_BG_PALETTE` instead of `DEFAULT_PALETTE`, in that case the shape controller could do substitutions differently if you specified something like `cc<1-5>=#XYZXYZ` instead of `c1=...`, but the default cc-bg palette `cc<1-5>` would just be a reordering of the `o-color-<1-5>` palette. I'm not sure the change in complexity is worth the very marginal increase in clarity, since basically all it changes is the way you think about which number you're putting in the shape declaration, I don't think it's too weird to just say "shape colors will be resolved to the corresponding cc-background color to allow them to match, which will not be exactly the same in a few palettes"

Reply:
> I still need to map from a color number (c1) to a different cc (c1 => cc4)

The point is that the "master code" uses the logic of a compatibility code, which seems very wrong. If a community module (or one of our future new theme) create a theme with color palettes and shapes, it will have to reverse your logic to know which strange number to add for its colors attribute of its new shapes.

Example: the community module declares its palette with "preset number 3 uses color number 4 by default" and one of the new shape also wants to use color number 4 as a default... the user has to understand it must add `colors: (1)` because of your hardcoded mapping... wtf, neither 3 or 4? :scream: `cc-bgs: (3)` is what a dev expects using the master API (and `colors: (4)` is what he expected in the compat API).

> if I only have the cc color I'd need to map that to the color number so that I can use it the the URL params.

I don't understand what is your issue. If the shape declare "I want to use the bg color of the cc number 4"... then you can send `o-color('o-cc4-bg')` in the URL... just like you did after the problematic mapping.

>  it also makes creating new shapes less natural: at the moment, the colors field in this dict is simply "which o-color-xs of the default palette does this shape use"

And so afterwards it will be "which background-color of ccx to use"... it seems as natural as before considering the need we have to use cc background colors. The point still being you cannot hardcode a mapping between ccx -> o-color-y which makes sense (see point 1).

I don't understand the rest of your comment, still seems in conflict with the hardcoding problem.

Reply:
Probably better we talk about it on Tuesday so I can show you what I mean.